metricflow/model/validations/validator_helpers.py

```diff
@@ -39,82 +40,121 @@ class ModelObjectType(Enum):
     METRIC = "metric"
 
 
+class ValidationContext(BaseModel):
+    """The base context class for vaidation issues"""
+
+    file_name: Optional[str]
+    line_number: Optional[int]
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+
+        context_string = ""
+
+        if self.file_name:
+            context_string += f"in file `{self.file_name}`"
+            if self.line_number:
+                context_string += f" on line #{self.line_number}"
+
+        return context_string
+
+
+class MaterializationContext(ValidationContext):
+    """The context class for vaidation issues involving materializations"""
+
+    materialization_name: str
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+        return f"with materialization `{self.materialization_name}` {ValidationContext.context_str(self)}"
+
+
+class MetricContext(ValidationContext):
+    """The context class for vaidation issues involving metrics"""
+
+    metric_name: str
+
```

Comment:
So above you are passing "materialization_name" and that doesn't blow up at either typecheck time or in the runtime because by default Pydantic will simply drop that key/value pair on the floor.